cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/Context/SchemaGenerator.ts

```diff
@@ -123,9 +123,10 @@ function generateEnumType(children: IOrderedChildren, currentNode: INode, nullab
 function generateRecordType(children: IOrderedChildren, currentNode: INode, nullable: boolean) {
   const finalType: IRecordField = {
     type: AvroSchemaTypesEnum.RECORD,
-    name: currentNode.name || `name-${uuidV4()}`,
+    name: currentNode.name || `name_${uuidV4()}`,
     fields: [],
   };
+  finalType.name = finalType.name.replace(/-/g, '');
```

Comment:
we might want to only do this replace is we were the one generating the name. If it's the user, such as a nested record schema that already have the currentNode.name, this might overwrite what they put, and there might be some database that actually support `-` in the column name

Reply:
Avro Schema specification doesn't allow `-`s in names. https://avro.apache.org/docs/current/spec.html#names. This is the record name. Not sure how this will be related to databases.